go/ct/gen/transaction_context_test.go

```diff
@@ -33,3 +94,205 @@ func TestTransactionContext_Generate(t *testing.T) {
 		t.Errorf("Generated origin address has default value.")
 	}
 }
+
+func TestTransactionContextGenerator_GenerateConstrained(t *testing.T) {
+
+	variableCheck := func(txCtx st.TransactionContext, assignment Assignment, t *testing.T,
+		variable Variable, shouldBePresent, shouldBeAssigned bool, value common.U256) {
+		t.Helper()
+		assignedValue, ok := assignment[variable]
+		if !ok {
+			t.Errorf("Variable %v should be in assignment.", variable.String())
+		}
+		if shouldBePresent {
+			if !assignedValue.IsUint64() {
+				t.Errorf("Variable %v should be assigned a uint64 value.", variable.String())
+			}
+			if assignedValue.Uint64() >= uint64(len(txCtx.BlobHashes)) {
+				t.Errorf("Assigned value for %v %v is out of range.", variable.String(), assignedValue.Uint64())
+			}
+		} else {
+			if assignedValue.Uint64() < uint64(len(txCtx.BlobHashes)) {
+				t.Errorf("Assigned value for %v %v is not out of range.", variable.String(), assignedValue.Uint64())
+			}
+		}
+		if shouldBeAssigned {
+			if !assignedValue.Eq(value) {
+				t.Errorf("Assigned value for %v %vis not the expected value.", variable.String(), assignedValue.Uint64())
+			}
+		}
```

Comment:
This last check seems to only make sure that what was assigned before the generator was executed remained present.

A more readable way to do this would be to
- remove this condition here and the two parameters `shouldBeAssigned` and `value` from this function
- in the test loop below, create a backup of the assignment before you pass it to the generator
- after the generator ran, check that the resulting assignment contains everything that is in the backup

This improved the readability of the `variableCheck` calls.